src/js/utils/ResourcesUtil.js

```diff
@@ -55,7 +55,13 @@ const ResourcesUtil = {
       }
     }
 
-    let resources = Object.keys(item.get("used_resources"));
+    let usedResources = item.get("used_resources");
+
+    if (!usedResources) {
+      usedResources = {};
```

Comment:
Hey, Michael!

In `structs/Node.js` we already guard against this case in one method like this 
```js
const value = (this.get("used_resources") || {})[resource];
``` 

Wouldn't it be better to create a getter method for `used_resources` that will always return an object use this getter method everywhere?

Reply:
I agree @nLight, I will address this.